subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java

```diff
@@ -120,22 +169,17 @@ public static synchronized void initialize(File userHomeDir, boolean initializeA
                             throw ex;
                         }
                     }
-                    if (initializeAdditionalNativeLibraries) {
-                        if (useNativeIntegrations) {
-                            useFileSystemWatching = true;
-                            try {
-                                FileEvents.init(nativeBaseDir);
-                            } catch (NativeIntegrationUnavailableException ex) {
-                                LOGGER.debug("Native file system watching is not available for this operating system.", ex);
-                                useFileSystemWatching = false;
-                            }
-                        }
-                        JANSI_BOOT_PATH_CONFIGURER.configure(nativeBaseDir);
-                    }
                     LOGGER.info("Initialized native services in: {}", nativeBaseDir);
                 }
                 initialized = true;
             }
+            for (NativeFeatures requestedFeature : requestedFeatures) {
+                if (!INITIALIZED_FEATURES.contains(requestedFeature)) {
+                    INITIALIZED_FEATURES.add(requestedFeature);
```

Comment:
I find this weird with capitals like this. Maybe we have too much static state here, and should have a static instance of a new thing that has instance fields instead?

Reply:
I moved most of the static fields to the actual `NativeServices` instance. I think this makes the code easier to understand.